### PR TITLE
Add VSTest Adapter and update BenchmarkDotNet

### PR DESCRIPTION
**Pull request by caaavik-msft**
This PR adds the new VSTest adapter for BenchmarkDotNet, allowing you to run benchmarks on your machine using your IDE's test runner. This also updates BenchmarkDotNet to the latest prerelease version. Will need to validate the BDN update does not cause other issues.

image

### CODE REVIEW

**Review by cincuranet — DISMISSED** [review]

**Review by LoopedBard3 — APPROVED** [review]
LGTM once merge conflict is fixed

**Pull request merged**
